torch/csrc/autograd/record_function.cpp

```diff
@@ -11,177 +12,192 @@ namespace profiler {
 
 namespace {
 
-float sample_zero_one() {
-  static thread_local auto gen =
-      torch::make_unique<std::mt19937>(std::random_device()());
-  std::uniform_real_distribution<float> dist(0.0, 1.0);
-  return dist(*gen);
+// Used to generate unique callback handles
+uint64_t next_unique_callback_id() {
```

Comment:
nit: can it be just an atomic<int>? :)

Reply:
there were some concerns voiced re. usage of atomics, but in the end record_function_benchmark will show